### ISSUE TITLE
[Add] Native Modules and NativeUIModules support

### ISSUE THREAD

**Issue by muskeinsingh** [necolas/react-native-web#1785]
**Is your feature request related to a problem? Please describe.**
React Native comes out with a first class support for Native Modules and Native UI Modules and that attempts to solve a LOT of use cases and bridge(pun intended) the gap between Native and the React world. I have a crude implementation of how these could work with the current architecture of RN(registering singleton instances of those modules pre-loading) within the web world too. 

**Describe a solution you'd like**
Basically, a way to expose NativeModules and NativeUIModules capability to web as well. 

**Describe alternatives you've considered**

**Additional context**
Not sure if this is being already worked on or if this doesn't align well with the future roadmap of the library but just in case it does, do let me know and I can help carve out a rusty PR with the same.

**Comment by necolas** [comment]
Some of that stuff might be changing in RN with Fabric. @EvanBacon has worked on a universal module format for Expo; he might have some input.

**Comment by muskeinsingh** [comment]
@EvanBacon can you share your insights here? could be useful to help decide the way forward.

**Comment by necolas** [comment]
Didn't you already do this at Flipkart? You could share a PR of that work to start a discussion

**Comment by muskeinsingh** [comment]
sure, I can share that. will ready a PR 👍

**Comment by rchavezj** [comment]
I'm very interested in native modules used for this hybrid library. So at this moment does react-native-web support ios-swift and java-android inside react-native?

**Comment by rchavezj** [comment]
I'm not completely sure what "Mocked. Missing ability to load native modules." means seeing the status of this feature has '(✓)'

**Comment by necolas** [comment]
It means the app won't crash from importing the `NativeModules` export. You shouldn't expect native code to run on the web

**Comment by necolas** [comment]
Any follow up here @muskeinsingh?

**Comment by necolas** [comment]
I reached out over DM to Flipkart eng management about this and other things, but they didn't follow through. Closing due to lack of follow-up.
